presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -923,8 +923,10 @@ private PlanWithProperties deriveProperties(PlanNode result, List<StreamProperti
         private PlanWithProperties accept(PlanNode node, StreamPreferredProperties context)
         {
             PlanWithProperties result = node.accept(this, context);
+            // TableWriter and TableWriterMergeNode has different output
+            boolean passStatsEquivalentPlanNode = !(node instanceof TableWriterNode && result.getNode() instanceof TableWriterMergeNode);
             return new PlanWithProperties(
-                    result.getNode().assignStatsEquivalentPlanNode(node.getStatsEquivalentPlanNode()),
+                    passStatsEquivalentPlanNode ? result.getNode().assignStatsEquivalentPlanNode(node.getStatsEquivalentPlanNode()) : result.getNode(),
```

Comment:
do we expect to see more false positives because of this? that is, plans that shouldn't match now match

Reply:
This is the only case I am aware of now, in which the result node has different number of rows compared with the original node.

Reply:
Can we do this when creating TableWriterMergeNode? Can we add a test for this?

Reply:
I tried but it seems non-trivial to change the logic when creating the TableWriterMergeNode. The accept method is the default one always called.

Added a test for it to make sure that no stats equivalent node is propagated to TableWriterMergeNode